src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -205,72 +206,129 @@ public static string DeterminePublishLocation(string workingDirectory, string pr
+                process.WaitForExit(5000);
 
-                // If it hasn't completed in the specified timeout, stop the process and give up
-                if (!hasExited) 
+                if (process.ExitCode == 0)
                 {
-                    process.Kill();
-                    msbuildProcessFailed = true;
-                }
+                    using JsonDocument doc = JsonDocument.Parse(output);
+                    JsonElement root = doc.RootElement;
+                    JsonElement propertiesElement = root.GetProperty("Properties");
 
-                // If it has completed but unsuccessfully, give up
-                if (process.ExitCode != 0)
+                    foreach (var property in propertyNames)
+                    {
+                        if (propertiesElement.TryGetProperty(property, out JsonElement propertyValue))
+                        {
+                            properties[property] = propertyValue.GetString();
+                        }
+                    }
+                }
+                else
                 {
-                    msbuildProcessFailed = true;
+                    Console.WriteLine($"MSBuild process exited with code {process.ExitCode}. Error: {error}");
+                    // Fallback to XML parsing
+                    properties = LookupProjectPropertiesFromXml(projectFile, propertyNames);
                 }
             }
-            catch (Exception)
+            catch (Exception ex)
             {
-                // swallow any exceptions related to `dotnet msbuild`
-                msbuildProcessFailed = true;
+                Console.WriteLine($"Error executing MSBuild or parsing output: {ex.Message}");
+                // Fallback to XML parsing
+                properties = LookupProjectPropertiesFromXml(projectFile, propertyNames);
             }
 
-            if (msbuildProcessFailed)
+            return properties;
+        }
+
+        private static Dictionary<string, string> LookupProjectPropertiesFromXml(string projectFile, string[] propertyNames)
+        {
+            var properties = new Dictionary<string, string>();
+            try
             {
-                var projectFile = FindProjectFileInDirectory(projectLocation);
                 var xdoc = XDocument.Load(projectFile);
-                var element = xdoc.XPathSelectElement("//PropertyGroup/OutputType");
-                output = element?.Value;
+                foreach (var propertyName in propertyNames)
+                {
+                    var element = xdoc.XPathSelectElement($"//PropertyGroup/{propertyName}");
+                    if (element != null && !string.IsNullOrWhiteSpace(element.Value))
+                    {
+                        properties[propertyName] = element.Value;
+                    }
+                }
             }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error parsing project file XML: {ex.Message}");
```

Comment:
No `Console.WriteLine` calls are used. This same code is used inside Visual Studio and there is no console there. If you want to log something you need to pass in an instance if `IToolLogger` to handle logging.

Reply:
Same comment about logging